portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js

```diff
@@ -121,6 +121,14 @@ const uploadReportAndSendNotification = async (req, res) => {
 
     const existingReport = existingReports[0];
 
+    if (existingReport.status !== UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED) {
+      return res
+        .status(500)
```

Comment:
Should this status be a `403`?

Reply:
The user will probably see the "Problem with service" or "Something went wrong" page independent of which error is thrown (...probably). It's more of a semantic thing for `portal-api` where a `500` suggests there was an internal server error, whereas this is not an internal server error and more of a "data exists in the wrong format" error (which shouldn't be a `403` either in this case so I'm not sure why I suggested that 😅 ). Either way, it's not like we handle anything differently depending on the error code, so there isn't much point of changing this - it was more of a curiosity thing!

Reply:
Yeah that makes sense, and I'm actually not sure semantically which it is! I think other than a user trying to access the endpoint directly when they shouldn't (defo a 4 something case!), the most likely (although probably not that likely) case for this is if a user loads the page for upload but then before uploading a different user uploads a report for the period, in which case I'm actually not sure what the correct response is. 500 doesn't seem too unreasonable, but perhaps 409 might technically be the correct choice?